### PR TITLE
rewrite readme

### PR DESCRIPTION
**Pull request by eristhea**
TODO:
- [x] make a guide on using enigma

rewrites the QM readme so it's not
a. incorrect
or
b. a weird homunculus of yarn's readme + the word "quilt" added on top

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
-Quilt Mappings uses Gradle to provide a number of utility tasks for working with the mappings.
+In order to get started with making your first pull request, 
+please see the guide in [CONTRIBUTING.md](CONTRIBUTING.md#guide-pull-requests)!
+
+## Gradle Tasks
+
+QM has a set of helpful Gradle tasks to make mapping easier, including:
 
 ### `mappings`
 Setup and download and launch the latest version of [Enigma](https://github.com/QuiltMC/Enigma) automatically configured to use the merged jar and the mappings.
```
Comment (supersaiyansubtlety) [inline README.md]:
```suggestion
Setup, download, and launch the latest version of [Enigma](https://github.com/QuiltMC/Enigma) automatically configured to use the merged jar and the mappings.
```

File: GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Editing mappings
+
+## Starting up
+
+After running a mappings task (either `mappings` or `mappingsUnpicked` via Gradle), you'll be greeted with this imposing Swing GUI:
+![an image of enigma's main window, devoid of life. the "deobfuscated classes", "obfuscated classes", and "structure" dockers are open on the sides, and nothing is displayed in the centre of the screen.](https://github.com/ix0rai/quilt-mappings/assets/66223394/fc569c2c-7b6e-4d22-9199-dd9b52afd8ca)
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
![An image of enigma's main window, devoid of life. The "deobfuscated classes", "obfuscated classes", and "structure" dockers are open on the sides, and nothing is displayed in the centre of the screen.](https://github.com/ix0rai/quilt-mappings/assets/66223394/fc569c2c-7b6e-4d22-9199-dd9b52afd8ca)
```
Or is capitalization unconventional in image descriptions for some reason?

File: GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+The first thing you want to draw your attention to is the left of the screen, where you'll find the "Obfuscated Classes" and "Deobfuscated Classes" dockers.
+Obfuscated Classes will contain classes where the top-level name remains unmapped, and "Deobfuscated Classes" the mapped ones.
+Alternatively, you can use the merged "Classes" docker to view all classes at once, by pressing the button on the far left.
+Dockers can be resized and their titles dragged around so you can find the best setup for you!
+
+All three dockers on the left serve the same purpose: showing you the classes in the project, just like you'd see in your IDE.
+Each class has a little icon to show how well-mapped it is: a green check for done, a yellow dot for in progress, and a red x for untouched.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
Each class has a little icon to show how completely it's mapped: a green check for done, a yellow dot for in progress, and a red x for untouched.
```
"well" to me strongly implies a measure of quality, not % completeness

File: GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+Double-click on packages and classes to open them. You can also right-click to see a context menu with some shortcuts to help you map faster!
+When you open a class, it will be decompiled and the source code shown in the main view in the centre.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
When you open a class, it will be decompiled and the source code shown in the main view in the center.
```
"center" is the more conventional American English spelling.

File: GUIDE.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+   automatically mapped by our [enigma plugin](https://github.com/QuiltMC/quilt-enigma-plugin). Don't be afraid of editing green or grey names,
+   there's always something better to find!
+2. The file is in a tab: Enigma supports having multiple files decompiled and ready to access at once. Nice!
+3. The "structure" docker on the right has updated! We'll explain how to use that soon.
+
+## Naming
+
+The process of naming boils down to four steps:
+1. Read the relevant code. The first step in coming up with a name should always be reading and brainstorming your own names.
+   For every type of token, you can use the "show calls" tool on right-click to find usages if you're lacking in code to assess.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
   For every type of token, you can use the "show calls" tool in the right-click menu to find usages if you're lacking in code to assess.
```
Makes it a little clearer imo.

File: GUIDE.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+2. Reread some code! Taking a look at the code with your name applied will help you get a good idea of the quality of your name.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]: *2. sounds like it should have an "apply name" step preceding it.

File: GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+### Tips: naming fields
+
+With fields, "show calls" is your best friend. You'll want to run through many of these before finalising your decision.
+Constants will often appear to have no usages, which is misleading: the compilation has inlined its value instead of using a reference.
+Use `ctrl + f` to find what you're looking for, and reference the Mojang name since Yarn will often fail to map these.
+
+### Tips: naming classes
+
+Class names should come with plenty of deliberation, given that they're the most visible of all.
+Make sure to use the "show inheritance" and "show implementations" tools on right-click to make sure the name you're choosing isn't too granular.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
Make sure to use the "show inheritance" and "show implementations" tools on right-click to make sure the name you're choosing is appropriately broad/narrow.
```
Not 100% on this, but seems like a more complete description of what to look out for imo.

File: GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+When considering package, either use an existing one or try to ensure there are 3+ classes that fit the name when creating a new package.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
When considering which package to put a class in, either use an existing one or try to ensure there are 3+ classes that fit the name when creating a new package.
```
I think this makes it a little clearer.

File: GUIDE.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+### Tips: naming methods
+
+With methods, the whole world of the context menu is open to you. Make sure to use all its options to sort through the references!
+Assume that the user knows the return type and the parameters when coming up with a name.
+
+### Tips: naming parameters
+
+Parameters can be tough to name, since Mojmap doesn't ship their names for us to reference. Be wary of differences between Yarn and QM making their name incompatible.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]: > Be wary of differences between Yarn and QM making their name incompatible.

Maybe elaborate on this, possibly with an example?

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]
One more thing: I hadn't heard the term "docker" used this way before. I'd heard of "docking" views to the side, though, so it was reasonably easy to figure out from context.

It might be worth having a short explanation of what docker means, but I'm not sure how widely known this term is so I can't really say for sure.

File: GUIDE.md (hunk 2 above)
Comment (eristhea) [reply GUIDE.md]: nope, I was just lazy on my capitalisation pass apparently

File: GUIDE.md [hunk 10]
```diff
@@ -42,7 +42,8 @@ Use `ctrl + f` to find what you're looking for, and reference the Mojang name si
 ### Tips: naming classes
 
 Class names should come with plenty of deliberation, given that they're the most visible of all.
-Make sure to use the "show inheritance" and "show implementations" tools on right-click to make sure the name you're choosing isn't too granular.
+Make sure to use the "show inheritance" and "show implementations" tools on right-click to make sure the name you're choosing is appropriately broad or narrow.
+To quote the example from the [conventions](CONVENTIONS.md), if a block class is only used to disable spawning, but has only one usage and no inheritors, it shouldn't be named `NoSpawningBlock`, it should be named after its usage: `BedrockBlock`.
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
To use the example from the [conventions](CONVENTIONS.md), if a block class is only used to disable spawning, but has only one usage and no inheritors, it shouldn't be named `NoSpawningBlock`, it should be named after its usage: `BedrockBlock`.
```
it's not actually quoted, just the same example is used.

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

**Review by supersaiyansubtlety — APPROVED** [review]

File: GUIDE.md [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+## Naming
+
+The process of naming boils down to four steps:
+1. Read the relevant code. The first step in coming up with a name should always be reading and brainstorming your own names.
+   For every type of token, you can use the "show calls" tool in the right-click menu to find usages if you're lacking in code to assess.
+2. Once you've applied your new name, reread some code! Taking a look at the code with the name you invented will help you get a good idea of the quality of your name.
+3. Cross-reference your name. QM mappers have access to Yarn and Mojmap to compare their names against, using tools like [Linkie](https://linkie.shedaniel.me/mappings)
+   and [WagYourTail's mapping viewer](https://wagyourtail.xyz/Projects/MinecraftMappingViewer/App). If any other name contradicts yours, triple-check your name to make
+   sure you haven't made a mistake! After checking the other names, hybridize, copy, or keep your original name to find the best possible option.
+4. Consider Javadoc. If your name is unclear, and you can't improve it without making it too wordy, throw in some Javadoc to make sure modders know
```
Comment (IotaBread) [inline GUIDE.md:33]: There isn't info in this document on how to add Javadoc to an entry

File: GUIDE.md [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+   what they're looking at.
+
+### Tips: naming fields
+
+With fields, "show calls" is your best friend. You'll want to run through many of these before finalising your decision.
+Constants will often appear to have no usages, which is misleading: the compilation has inlined its value instead of using a reference.
```
Comment (IotaBread) [inline GUIDE.md]:
```suggestion
Constants (`static final` fields)  will often appear to have no usages, which is misleading: the compilation has inlined its value instead of using a reference.
```

File: GUIDE.md
Comment (IotaBread) [inline GUIDE.md:1]: Might be worth adding a "Found a bug?" section

File: GUIDE.md [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+to help you find unmapped items and examine the class. The entries in the tree can be double-clicked to navigate to them.
+
+### Inheritance
+
+When you use "show inheritance" in the context menu, this docker creates a tree showing you what it's found.
+The entries in the tree can be double-clicked to navigate to them.
+
+### Calls
+
+Similarly to the Inheritance docker, the Calls docker shows a tree of found calls when you use the "show calls" tool.
```
Comment (IotaBread) [inline GUIDE.md:102]:
```suggestion
Similarly to the Inheritance docker, the Calls docker shows a tree of found calls when you use the "show calls" tool. Methods with more than a single call within a same caller method will show a list of the lines where they are, in a panel just below the main calls list
```

**Review by IotaBread — APPROVED** [review]

File: GUIDE.md (hunk 13 above)
Comment (supersaiyansubtlety) [reply GUIDE.md:102]: > within a same caller method

within *the* same caller method?

Comment (IotaBread) [reply GUIDE.md:102]: oops, yeah
```suggestion
Similarly to the Inheritance docker, the Calls docker shows a tree of found calls when you use the "show calls" tool. Methods with more than a single call within the same caller method will show a list of the lines where they are, in a panel just below the main calls list
```

File: GUIDE.md
Comment (eristhea) [reply GUIDE.md:1]: love that our program is so broken we need this

File: GUIDE.md [hunk 14]
```diff
@@ -99,7 +99,7 @@ The entries in the tree can be double-clicked to navigate to them.
 
 ### Calls
 
-Similarly to the Inheritance docker, the Calls docker shows a tree of found calls when you use the "show calls" tool.
+Similarly to the Inheritance docker, the Calls docker shows a tree of found calls when you use the "show calls" tool. Methods with more than a single call within the same caller method will show a list of the lines where they are, in a panel just below the main calls list
```
Comment (supersaiyansubtlety) [inline GUIDE.md]:
```suggestion
Similarly to the Inheritance docker, the Calls docker shows a tree of found calls when you use the "show calls" tool. Methods with more than a single call within the same caller method will show a list of the lines where they are, in a panel just below the main calls list.
```

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

**Comment by eristhea** [comment]
@supersaiyansubtlety you're too fast I was already fixing that :p

**Review by supersaiyansubtlety — APPROVED** [review]

**Review by IotaBread — APPROVED** [review]

**Review by EnnuiL — COMMENTED** [review]
hm, before I review this tomorrow, I'm wondering: shouldn't we get rid of those line breaks in favor of text wrapping? I feel like text wrapping on IDE's side might end being better for like, writing and expanding these docs

File: GUIDE.md [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+
+### Themes
+
+Enigma supports customising its theme, with a few built-in options. Yes, this includes a dark mode.
+You can change the theme in the menu bar, under `Theme`.
+
+## Found a bug?
+
+Enigma in its current state is a famously broken piece of software, and we're working hard to fix it!
+If you find the slightest bug or have an idea to improve it, please submit it on [our issue tracker](https://github.com/quiltmc/enigma/issues).
```
Comment (EnnuiL) [inline GUIDE.md]:
```suggestion
If you find the slightest bug or have an idea to improve it, please submit it on [our issue tracker](https://github.com/QuiltMC/enigma/issues).
```

File: README.md [hunk 16]
```diff
@@ -1,37 +1,58 @@
 # Quilt Mappings
 
-Quilt Mappings is a set of open, unencumbered Minecraft mappings, free for everyone to use under the Creative Commons Zero license. The intention is to let 
-everyone mod Minecraft freely and openly, while also being able to innovate and process the mappings as they see fit.
+Quilt Mappings is a set of open, accurate, and easy-to-understand Minecraft mappings
+under the Creative Commons Zero license. Quilt Mappings seeks to provide the best of
+both worlds in terms of mapping quality: the accuracy and completeness of Mojang's
+official mappings, and the simplicity and set-in-stone conventions of its basis, the
+[Yarn](https://github.com/fabricmc/yarn) project.
```
Comment (EnnuiL) [inline README.md]:
```suggestion
[Yarn](https://github.com/FabricMC/yarn) project.
```

File: README.md [hunk 17]
```diff
@@ -1,37 +1,58 @@
...
+To see what version is currently being targeted for updates, check the current Git branch's name.
 
-## Usage
-To use Quilt Mappings-deobfuscated Minecraft for Minecraft modding or as a dependency in a Java project, you can use [quilt-gradle](https://github.com/quiltmc/quilt-gradle) Gradle plugin. See [fabric wiki tutorial](https://fabricmc.net/wiki/tutorial:setup) for more information.
+## Using QM
 
-To obtain a deobfuscated Minecraft jar, [`./gradlew mapNamedJar`](#mapNamedJar) will generate a jar named like `<minecraft version>-named.jar`, which can be sent to a decompiler for deobfuscated code.
+To use Quilt Mappings in your mod, you can use either [quilt-loom](https://github.com/quiltmc/quilt-loom)
+for Quilt mods or use our Intermediary publication on [Fabric's loom](https://github.com/fabricmc/fabric-loom)
+for Fabric mods. We recommend cloning the [Quilt Template Mod](https://github.com/quiltmc/quilt-template-mod)
```
Comment (EnnuiL) [inline README.md]:
```suggestion
To use Quilt Mappings in your mod, you can use either [quilt-loom](https://github.com/QuiltMC/quilt-loom)
for Quilt mods or use our Intermediary publication on [Fabric's loom](https://github.com/FabricMC/fabric-loom)
for Fabric mods. We recommend cloning the [Quilt Template Mod](https://github.com/QuiltMC/quilt-template-mod)
```

File: README.md [hunk 18]
```diff
@@ -1,37 +1,58 @@
...
+In order to get started with making your first pull request, 
+please see the guide in [CONTRIBUTING.md](CONTRIBUTING.md#guide-pull-requests)!
+
+## Gradle Tasks
+
+QM has a set of helpful Gradle tasks to make mapping easier, including:
 
 ### `mappings`
-Setup and download and launch the latest version of [Enigma](https://github.com/QuiltMC/Enigma) automatically configured to use the merged jar and the mappings.
+Setup, download, and launch the latest version of [Enigma](https://github.com/QuiltMC/Enigma) automatically configured to use the merged jar and the mappings.
```
Comment (EnnuiL) [inline README.md]:
```suggestion
Setup, download, and launch the latest version of [Enigma](https://github.com/QuiltMC/enigma) automatically configured to use the merged jar and the mappings.
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]
oh i can tackle this small bit today

**Review by OroArmor — APPROVED** [review]
Looks good with Ennui's changes

**Comment by eristhea** [comment]
I Hate Capital Letters

**Review by supersaiyansubtlety — APPROVED** [review]

**Review by EnnuiL — APPROVED** [review]
